### PR TITLE
Avoid risky truthy-falsy comparison

### PR DESCRIPTION
**Pull request by greg0ire**
Stumbled upon this while working on https://github.com/doctrine/collections/pull/386

### CODE REVIEW

File: src/ArrayCollection.php [hunk 1]
```diff
@@ -480,7 +480,7 @@ public function matching(Criteria $criteria)
         $offset = $criteria->getFirstResult();
         $length = $criteria->getMaxResults();
 
-        if ($offset || $length) {
+        if ($offset !== null && $offset > 0 || $length !== null && $length > 0) {
             $filtered = array_slice($filtered, (int) $offset, $length, true);
```
Comment (SenseException) [inline src/ArrayCollection.php:484]:
```suggestion
        if ($offset !== null && $offset > 0 || $length !== null && $length > 0) {
            $filtered = array_slice($filtered, $offset, $length, true);
```

**Review by SenseException — CHANGES_REQUESTED** [review]
Is this PR a cosmetic one?

**Comment by greg0ire** [comment]
> Is this PR a cosmetic one?

Sorry, I should have mentioned that it fixes the build by addressing an issue found by Psalm

**Comment by greg0ire** [comment]
@SenseException after applying your suggestion the tests break like this:

```
1) Doctrine\Tests\Common\Collections\ArrayCollectionTest::testMatchingWithSlicingPreserveKeys with data set "preserve keys on maxresult only" (array(1, 2, 3, 4), array(1, 2), null, 2)
TypeError: array_slice(): Argument #2 ($offset) must be of type int, null given

/home/greg/dev/doctrine-collections/src/ArrayCollection.php:484
/home/greg/dev/doctrine-collections/tests/Common/Collections/BaseArrayCollectionTest.php:356
```

**Review by SenseException — APPROVED** [review]
Because of the `||` it can be null. I forgot that case is still valid.

**Pull request merged**